app/controllers/order/schedule_details.js

```diff
@@ -152,15 +152,21 @@ export default Controller.extend(cancelOrder, {
 
       new AjaxPromise(url, actionType, this.get("session.authToken"), {
         order_transport: this.orderTransportParams()
-      }).then(data => {
-        this.get("store").pushPayload(data);
-        loadingView.destroy();
-        if (previousRouteName === "orders.booking") {
-          this.transitionToRoute(previousRouteName, orderId);
-        } else {
-          this.transitionToRoute("order.confirm_booking", orderId);
-        }
-      });
+      })
+        .then(data => {
+          this.get("store").pushPayload(data);
+          loadingView.destroy();
+          if (previousRouteName === "orders.booking") {
+            this.transitionToRoute(previousRouteName, orderId);
+          } else {
+            this.transitionToRoute("order.confirm_booking", orderId);
+          }
+        })
+        .catch(err => {
+          this.get("messageBox").alert(err.responseJSON.error, () =>
+            this.transitionToRoute("/")
```

Comment:
Does it make sense to transition back to the order or list of orders (my dashboard) rather than the home page?